### PR TITLE
AWS High Availability Architecture

### LINKED ISSUE #23: AWS High Availability Architecture
**MicroByte Title:**  AWS High Availability Architecture

**Domain:**  Web Service

**Scope of learning for learners:**  
- Learner will be able to deploy a website with `High availability`, `secure`  website, and make website `manageable`.

**Activities:** 
- Activity 1 (Launched an instance in AWS): launched an instance in AWS using AWS CLI and this activity contain task for the learner.
- Activity 2 (Configured Webserver): configured Webserver by installing Apache Webserver and this activity contain task for the learner.
- Activity 3 (Created an EBS Volume): created EBS Volume using AWS CLI.
- Activity 4 (Attach EBS Volume to Instance): attached EBS Volume to the instance using AWS CLI and this activity contain task for the learner.
- Activity 5 (Created and Formated a Partition): created a Partition in Linux instance and then format the partition.
- Activity 6 (Mounted /var/www/html directory to partition): mounted  /var/www/html directory to the partition and created a HTML Page and this activity contain task for the learner.
- Activity 7 (Created S3 Bucket): created S3 bucket using AWS CLI.
- Activity 8 (Uploaded images in S3 Bucket):Uploaded images in S3 bucket and copy the URL of images and pasted in webpage.
- Activity 9(Created Cloudfront Distribution ): Created Cloudfront Distribution using AWS CLI and this activity contain task for the learner.

### PR DESCRIPTION
**Pull request by Nitesh-thapliyal**
### Checklist: 

- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Comment by amoghrajesh** [comment]
Re - Earlier Issues:
Hey @Nitesh-thapliyal!
You have made most of the changes requested by me, but you might have missed out a few so I am gonna reiterate it for you:

In Prerequisites, you are assuming that it is a Windows user. What if he uses Linux/Mac? Provide links for these too.
You still have not mentioned what you need in the HTML in activity 7. Tell the learner how big the html should be (Just a simple boilerplate or does it need to have anything)
You got rid of the part in the previous version when you were launching the website in terms of activity 11 (Add that part but in Activity 10).
Apart from all these changes, very good work, man.
Makes these changes as soon as possible.

File: AWS High Availability Architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  width: 100%;
+}
+</style>
+</head>
+<body>
+<img src="html5.png"  width="128" height="128">
+<img src="html5.png" width="128" height="128">
+</body>
+</html> 
+```
```
Comment (amoghrajesh) [inline AWS High Availability Architecture/README.md:167]: Just add here that: "Create an HTML file with your own content or if you can't think of content, use this one as below"

File: AWS High Availability Architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+**Activity 9 Task:**
+
+Find S3 Bucket domain using AWS CLI or WebUi.
+
+Now open the webpage using the Ip address of instance and file name in your browser .
+
+> Refer to `Images/Activity 9/Webpage.png`
+
+
+By performig all Activities you can deploy your website with `High availability`,`secure` your website and make your website `manageable`.
```
Comment (amoghrajesh) [inline AWS High Availability Architecture/README.md]: performing* - typo

File: AWS High Availability Architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+- We will Create Partition
+- We will create S3 Bucket
+- We will Create Cloudfront Distribution
+
+I hope you will learn by doing the Activities and will perform Task given in the Activities #LearnByDoing
+
+# Prerequisites
+
+- You should have AWS account, you can create [AWS free tier account](https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc).
+- Install AWS CLI in [Windows](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-windows.html) or in [Mac](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-mac.html).
```
Comment (amoghrajesh) [inline AWS High Availability Architecture/README.md]: What about Linux?

File: AWS High Availability Architecture/README.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
+# Introduction
+
+![AWS](https://blogs.bmc.com/wp-content/uploads/2020/06/regions.png)
```
Comment (amoghrajesh) [inline AWS High Availability Architecture/README.md]: What does this image mean? Add a caption for it

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @Nitesh-thapliyal!
This is pretty cool content and I like it personally.
I have requested very minor changes and after that, it is good to go from my end.
Thanks for your generous contribution & let me know once you make these.

**Review by amoghrajesh — APPROVED** [review]
@Nitesh-thapliyal perfect!
Thanks for your quick response to editing your work.
I like this content. Let's see what other reviewers have to say about this one.

P.S: You missed out on my last comment in the previous review. Correct it till other reviewers see this one.
Also, start to think of some learner issues if possible that can be raised here.

Thanks!

File: AWS High Availability Architecture/README.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+A highly available architecture involves multiple components working together to ensure uninterrupted service during a specific period. This also includes the response time to users’ requests. Namely, available systems have to be not only online, but also responsive.
+
+Implementing a cloud computing architecture that enables this is key to ensuring the continuous operation of critical applications and services. They stay online and responsive even when various component failures occur or when a system is under high stress.
+
+Highly available systems include the capability to recover from unexpected events in the shortest time possible. By moving the processes to backup components, these systems minimize downtime or eliminate it. This usually requires constant maintenance, monitoring, and initial in-depth tests to confirm that there are no weak points.
+
+**Application:** If you want to deploy a website and want's your website to be fast, secure and you want your website to be available all the time even in the time of some failure in the network then`High Availabilty Architecture`comes in play.
```
Comment (Kevinpaulose05) [inline AWS High Availability Architecture/README.md]: Typo- want's should be want(or something) in line 13

Comment (YourAKShaw) [reply AWS High Availability Architecture/README.md]: If you want to deploy a website and also want it to be fast, secure and available all the time, even in the time of some failure in the network, then `High Availability Architecture` comes in play.

Just copy paste this instead.

File: AWS High Availability Architecture/README.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+- Install AWS CLI in [Windows](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-windows.html),[Mac](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-mac.html) or in [Linux](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html).
+- Configure [AWS CLI with IAM user](https://bauddhik-geek.blogspot.com/2020/10/working-in-aws-using-aws-cli-with-help.html).
+- It is preferred that you are aware [HTML basics](https://learn.crio.do/home/me/ME_HTML_CSS) and [Linux](https://learn.crio.do/home/me/ME_LINUX1)
+
+The Activities that we are going to perform on AWS are free in `AWS free tier plan`but you should know the limitation of free tier plan so, go to the [link](https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc) and check limitation.
+
+### Activity 1: Launch an Instance using AWS CLI
+First configure the AWS CLI in your [window cmd](https://www.online-tech-tips.com/computer-tips/how-to-use-dos-command-prompt/) by providing Access Key, Secret Key and region name.
```
Comment (YourAKShaw) [inline AWS High Availability Architecture/README.md]: A learner who might be using Linux/Mac-OS would have a terminal. It seems that is instruction will be similar for terminal too, kindly write it that way. Writing only `window cmd` might be a little misleading, plus it's `Windows`.

**Review by YourAKShaw — CHANGES_REQUESTED** [review]
Hi @Nitesh-thapliyal! Overall, the content seems well laid out. I'm impressed. Kindly make the other small changes quick. And we'll soon give it a parting approval, merging it.

**Review by Kevinpaulose05 — APPROVED** [review]
Awesome job @Nitesh-thapliyal . Just one suggestion - It would be better if u could centre align the first image (of the world map). Two things to do **after merging**(u already know this :) ) - 
1. Open issues for learners so that they can contribute to Ur microbyte
2. Showcase Ur microbyte in slack's microbyte showcase channel using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md)

Happy Contributing!!

File: AWS High Availability Architecture/README.md (hunk 6 above)
Comment (Kevinpaulose05) [reply AWS High Availability Architecture/README.md]: U can mention that the environment u have used is `windows` and so some things maybe be different for different OS

**Review by YourAKShaw — APPROVED** [review]

**Comment by YourAKShaw** [comment]
@all-contributors kindly add @Nitesh-thapliyal for:

- [x] code
- [x] content
- [x] ideas
- [x] doc

**Pull request merged**